src/Auth0.AuthenticationApi/Builders/LogoutUrlBuilder.cs

```diff
@@ -1,54 +1,70 @@
+        public LogoutUrlBuilder(string baseUrl)
             : base(baseUrl, "v2/logout")
         {
         }
 
         /// <summary>
-        /// Adds a returnTo query string parameter.
+        /// Initializes a new instance of the <see cref="LogoutUrlBuilder"/> class.
         /// </summary>
-        /// <param name="url">The URL.</param>
-        /// <returns>The <see cref="LogoutUrlBuilder"/>.</returns>
-        public LogoutUrlBuilder WithReturnUrl(string url)
+        /// <param name="baseUrl">Base URL of the Authentication API represented as a <see cref="Uri"/>.</param>
+        public LogoutUrlBuilder(Uri baseUrl)
+            : base(baseUrl, "v2/logout")
         {
-            AddQueryString("returnTo", url);
+        }
 
-            return this;
+        /// <summary>
+        /// Adds the `redirect_uri` query string parameter specifying the redirect URI.
+        /// </summary>
+        /// <param name="uri">URI to redirect to.</param>
+        /// <returns>Current <see cref="LogoutUrlBuilder"/> to allow fluent configuration.</returns>
+        public LogoutUrlBuilder WithReturnUrl(string uri)
+        {
+            return WithValue("returnTo", uri);
         }
 
         /// <summary>
-        /// Adds a client_id query string parameter.
+        /// Adds the `redirect_uri` query string parameter specifying the redirect URI.
         /// </summary>
-        /// <param name="clientId">The URL.</param>
-        /// <returns>The <see cref="LogoutUrlBuilder"/>.</returns>
-        public LogoutUrlBuilder WithClientId(string clientId)
+        /// <param name="uri"><see cref="Uri"/> to redirect to.</param>
+        /// <returns>Current <see cref="LogoutUrlBuilder"/> to allow fluent configuration.</returns>
+        public LogoutUrlBuilder WithReturnUrl(Uri uri)
         {
-            AddQueryString("client_id", clientId);
+            return WithReturnUrl(uri.OriginalString);
+        }
 
-            return this;
+        /// <summary>
+        /// Adds the `client_id` query string parameter specifying the Client ID of the application.
+        /// </summary>
+        /// <param name="clientId">Client ID of the application.</param>
+        /// <returns>Current <see cref="LogoutUrlBuilder"/> to allow fluent configuration.</returns>
+        public LogoutUrlBuilder WithClientId(string clientId)
+        {
+            return WithValue("client_id", clientId);
         }
 
         /// <summary>
-        /// Adds a federated query string parameter.
+        /// Adds the `federated` flag query string parameter (no value).
         /// </summary>
-        /// <returns>The <see cref="LogoutUrlBuilder"/>.</returns>
+        /// <returns>Current <see cref="LogoutUrlBuilder"/> to allow fluent configuration.</returns>
         public LogoutUrlBuilder Federated()
         {
-            AddQueryString("federated", null);
-
-            return this;
+            return WithValue("federated", null);
```

Comment:
will a `null` value mean the query param is included in the final string url that gets constructed? if not, you could set it to `1` to trigger that

Reply:
It just means it will add `?federated` without the `=` part which any other value would cause.